packages/overlays/src/OverlayMixin.js

```diff
@@ -147,8 +147,8 @@ export const OverlayMixin = dedupeMixin(
         if (this._overlayCtrl) {
           this.__tornDown = true;
           this.__overlayContentNodeWrapperBeforeTeardown = this._overlayContentNodeWrapper;
-          this._teardownOverlayCtrl();
         }
+        this._teardownOverlayCtrl();
```

Comment:
We tear down regardless of whether there is an overlay controller yet, because we setup the open and close event listeners on connectedCallback, which happens before setting up the overlay controller. So it can (and does) happen where the element gets connected, but no overlay controller is created before being disconnected.